File: src/map/MyHashSet.java
package map;

import java.util.LinkedList;

public class MyHashSet {

    public static void main(String[] args) {
        MyHashSet obj = new MyHashSet();
        int key = 6;
        obj.add(key);
        System.out.println(obj.contains(key));
        obj.remove(key);
        boolean param_3 = obj.contains(key);
        System.out.println(obj.contains(key));
    }

    private Bucket[] bucketArray;
    private int keyRange;

    /** Initialize your data structure here. */
    public MyHashSet() {
        this.keyRange = 769;
        this.bucketArray = new Bucket[this.keyRange];
        for (int i = 0; i < this.keyRange; ++i)
            this.bucketArray[i] = new Bucket();
    }

    protected int _hash(int key) {
        return (key % this.keyRange);
    }

    public void add(int key) {
        int bucketIndex = this._hash(key);
        this.bucketArray[bucketIndex].insert(key);
    }

    public void remove(int key) {
        int bucketIndex = this._hash(key);
        this.bucketArray[bucketIndex].delete(key);
    }

    /** Returns true if this set contains the specified element */
    public boolean contains(int key) {
        int bucketIndex = this._hash(key);
        return this.bucketArray[bucketIndex].exists(key);
    }

    class Bucket {
        private LinkedList<Integer> container;

        public Bucket() {
            container = new LinkedList<Integer>();
        }

        public void insert(Integer key) {
            int index = this.container.indexOf(key);
            if (index == -1) {
                this.container.addFirst(key);
            }
        }

        public void delete(Integer key) {
            this.container.remove(key);
        }

        public boolean exists(Integer key) {
            int index = this.container.indexOf(key);
            return (index != -1);
        }
    }

}

File: p_src/myhashset.py

class MyHashSet:
    def __init__(self):
        self.range = 769
        self.hashset = [Bucket() for _ in range(self.range)]

    def _hash(self, value):
        return value % self.range

    def add(self, value):
        index = self._hash(value)
        self.hashset[index].insert(value)

    def remove(self, value):
        index = self._hash(value)
        self.hashset[index].delete(value)

    def contains(self, value):
        index = self._hash(value)
        return self.hashset[index].exists(value)


class Node:
    def __init__(self, value):
        self.value = value
        self.next = None


class Bucket:
    def __init__(self):
        self.head = Node(0)

    def insert(self, value):
        if not self.exists(value):
            new_node = Node(value)
            curr = self.head
            while curr.next is not None:
                curr = curr.next
            curr.next = new_node

    def delete(self, value):
        prev = self.head
        curr = self.head.next
        while curr is not None:
            if curr.value == value:
                # remove the current node
                prev.next = curr.next
                return
            prev = curr
            curr = curr.next

    def exists(self, value):
        curr = self.head.next
        while curr is not None:
            if curr.value == value:
                return True
            curr = curr.next
        return False


if __name__ == '__main__':
    # Your MyHashSet object will be instantiated and called as such:
    key = 100
    obj = MyHashSet()
    obj.add(key)
    print(obj.contains(key))
    obj.remove(key)
    print(obj.contains(key))
